src/tools/crawler/main.rs

```diff
@@ -113,11 +113,17 @@ async fn main() {
 
             crawler.send_broadcast(Message::GetAddr).unwrap();
 
-            // Create summary and log to file.
-            let network_summary =
-                NetworkSummary::new(crawler.known_network.nodes(), crawler_start_time);
-            network_summary.log_to_file().unwrap();
-            info!("{}", network_summary);
+            if crawler.known_network.num_connections() > 0 {
+                // Create summary and log to file.
+                let network_summary = NetworkSummary::new(
+                    crawler.known_network.nodes(),
+                    crawler.known_network.connections(),
+                    crawler_start_time,
+                );
```

Comment:
TODO: we should probably do the following next:
- put `crawler_start_time` inside the `Crawler`
- make `NetworkSummary::new` take `&Crawler` as the only parameter